src/test/smart-components/user/addUserToGroup.test.js

```diff
@@ -0,0 +1,115 @@
+import React from 'react';
+import { Provider, useSelector } from 'react-redux';
+import { render, screen, waitFor } from '@testing-library/react'; 
+import { MemoryRouter, useHistory} from 'react-router-dom';
+import { addNotification } from '@redhat-cloud-services/frontend-components-notifications/';
+import AddUserToGroup from '../../../smart-components/user/add-user-to-group/add-user-to-group';
+import * as groupActions from '../../../redux/actions/group-actions';
+import userEvent from '@testing-library/user-event';
+import configureMockStore from 'redux-mock-store';
+import thunk from 'redux-thunk';
+import { FETCH_GROUPS, ADD_MEMBERS_TO_GROUP } from '../../../redux/action-types.js';
+
+const mockStore = configureMockStore([thunk]);
+const initialState = {
+  groupReducer: {
+    groups: {
+      data: [],
+      isLoading: false,
+    },
+  },
+};
+
+const renderComponent = (userName = 'testuser', store) => {
+  return render(
+    <Provider store={store}>
+      <MemoryRouter>
+        <AddUserToGroup username={userName} />
+      </MemoryRouter>
+    </Provider>
+  );
+};
+
+const testGroups = [
+  {
+     uuid: '1',
+     name: 'Group 1',
+     description: "The best group",
+  },
+  {
+    uuid: '2',
+    name: 'Group 2',
+    description: 'The second best',
+  }
+];
+
+describe('Add User to Group Wizard', () => {
+  let fetchGroups, addMembersToGroup, store;
+  
+  beforeEach(() => {
+    jest.clearAllMocks();
+  
+    fetchGroups = jest.spyOn(groupActions, 'fetchGroups');
+    addMembersToGroup = jest.spyOn(groupActions, 'addMembersToGroup');
+
+    store = mockStore({
+      ...initialState,
+      groupReducer: {
+        groups: {
+          data: testGroups,
+          isLoading: false,
+        },
+      },
+    });
+
+    fetchGroups.mockImplementationOnce(() => ({
+     type: FETCH_GROUPS,
+      payload: Promise.resolve({
+        value: {
+          data: testGroups,
+          meta: {
+            count: testGroups.length,
+            limit: 10,
+            offset: 0,
+          },
+        },
+      }),
+    }));
+   
+  });
+
+  test('Add to group wizard renders correctly', () => {
+    const store = mockStore(initialState);
+
+    renderComponent('testuser', store);
+    expect(screen.getByText(/Add to group/i)).toBeInTheDocument();
```

Comment:
Can we also check the warning "This group list has been filtered to only show groups that the user is not currently a member of." is visible? We can directly reference the intl message

Reply:
I think we can test both :slightly_smiling_face: